### ISSUE TITLE
странность с попыткой переопределения

### ISSUE THREAD

**Issue by ArkadiyAliev** [GassaFM/interpr#29]
При повторном определении переменной, но уже с другим типом, 2 копии с одним именем существуют одновременно. Например в такой программе выводом будет 1 2
```
function sum (id, pr, n, a):
      f:=array(20)
      f:=1
      print(f)
      f[4]:=f+1
      print(f[4])
```
Получается, что существует 2 места, где можно встретить переменную с одним и тем же именем и интерпретатор не замечает, что одна из них уже существует. Например, в такой программе 
```
function sum (id, pr, n, a):
	f:=array(20)
        g:=1
	g:=f 
```
интерпретатор выдаст `no such variable: f`. А если добавить строчку `f:=1`, программа заработает

**Comment by GassaFM** [comment]
Да, всё так. Есть два пространства имён: одно для целочисленных переменных, другое для массивов. Поскольку из контекста интерпретатору всегда понятно, массив или целочисленная переменная имеется в виду, это ничему не противоречит.

А что предлагается сделать?

**Comment by ArkadiyAliev** [comment]
как вариант, запретить так делать, ведь далее если добавить какие то общие операции, например присвоение массивов те f:=g, возможно интерпретатору будет тяжелее

**Comment by GassaFM** [comment]
Присваивания массивов нет по делу: в #19 написано, почему.
Можно и запретить, но поскольку было незачем -- на это не тратилось время разработки.
